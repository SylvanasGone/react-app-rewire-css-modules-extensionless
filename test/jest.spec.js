'use strict';

const serializerPath = require('jest-serializer-path');
const rewireCssModulesExtensionless = require('..');

expect.addSnapshotSerializer(serializerPath);

const mockConfig = {
    transform: {
        '^.+\\.(js|jsx|mjs)$': 'path/to/config/jest/babelTransform.js',
        '^.+\\.css$': 'path/to/config/jest/cssTransform.js',
        '^.+\\.(graphql)$': 'path/to/config/jest/graphqlTransform.js',
        '^(?!.*\\.(js|jsx|mjs|css|json|graphql)$)': 'path/to/config/jest/fileTransform.js',
    },
    transformIgnorePatterns: [
        '[/\\\\]node_modules[/\\\\].+\\.(js|jsx|mjs)$',
    ],
    moduleNameMapper: {
        '^react-native$': 'react-native-web',
    },
};

it('should modify the jest config correctly', () => {
    const result = rewireCssModulesExtensionless.jest(mockConfig);

    expect(result).toMatchSnapshot();
});

it('should allow usage with compose', () => {
    expect(rewireCssModulesExtensionless.jest(mockConfig))
    .toEqual(rewireCssModulesExtensionless.jest()(mockConfig));
});
