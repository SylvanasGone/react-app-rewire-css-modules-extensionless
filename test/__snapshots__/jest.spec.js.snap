// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should modify the jest config correctly 1`] = `
Object {
  "moduleNameMapper": Object {
    "/.css$": "<PROJECT_ROOT>/node_modules/identity-obj-proxy/src/index.js",
    "^react-native$": "react-native-web",
  },
  "transform": Object {
    "^(?!.*/.(js|jsx|mjs|css|json|graphql)$)": "path/to/config/jest/fileTransform.js",
    "^.+/.(graphql)$": "path/to/config/jest/graphqlTransform.js",
    "^.+/.(js|jsx|mjs)$": "path/to/config/jest/babelTransform.js",
  },
  "transformIgnorePatterns": Array [
    "[///]node_modules[///].+/.(js|jsx|mjs)$",
  ],
}
`;
